src/main/java/me/dinowernli/grpc/prometheus/Configuration.java

```diff
@@ -32,23 +37,34 @@ public static Configuration allMetrics() {
     return new Configuration(
         true /* isIncludeLatencyHistograms */,
         CollectorRegistry.defaultRegistry,
-        DEFAULT_LATENCY_BUCKETS);
+        DEFAULT_LATENCY_BUCKETS,
+        new ArrayList<>());
   }
 
   /**
    * Returns a copy {@link Configuration} with the difference that Prometheus metrics are
    * recorded using the supplied {@link CollectorRegistry}.
    */
   public Configuration withCollectorRegistry(CollectorRegistry collectorRegistry) {
-    return new Configuration(isIncludeLatencyHistograms, collectorRegistry, latencyBuckets);
+    return new Configuration(isIncludeLatencyHistograms, collectorRegistry, latencyBuckets, labelHeaders);
   }
 
   /**
    * Returns a copy {@link Configuration} with the difference that the latency histogram values are
    * recorded with the specified set of buckets.
    */
   public Configuration withLatencyBuckets(double[] buckets) {
-    return new Configuration(isIncludeLatencyHistograms, collectorRegistry, buckets);
+    return new Configuration(isIncludeLatencyHistograms, collectorRegistry, buckets, labelHeaders);
+  }
+
+  /**
+   * Returns a copy {@link Configuration} that recognizes the given list of header names and uses their value from
```

Comment:
It took me a few reads through the code to fully grasp what this feature does. Could we have some more elaborate documentation here? For instance, adding an example here would be useful, as well as describing how this interacts with the standard labels, how sanitization works, etc.

Separately, we probably want some documentation in the top-level readme for the new features (and, as mentioned in the other comment, it would be great to make the feature a separate PR).

Reply:
Sounds good. I will add code docs and readme parts in the next PR.